tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,445 @@
+import sys
+import os
+
+
+FOURSPACES=@SPACES@
+NEWLINE=@NEWLINE@
+
+input_json_path = sys.argv[1]
+
+
+params = json.load(open(input_json_path, "r"))
+
+def write_images():
+  filter_images = params['images']['filter_images']
+
+  f.write(NEWLINE)
+  f.write("Images:[module_num:1|svn_version:\\'Unknown\\'|variable_revision_number:2|show_window:False|notes:\\x5B\\'To begin creating your project, use the Images module to compile a list of files and/or folders that you want to analyze. You can also specify a set of rules to include only the desired files in your selected folders.\\'\x5D|batch_state:array(\x5B\x5D, dtype=uint8)|enabled:True|wants_pause:False]"+NEWLINE)
+
+  f.write(FOURSPACES+":"+NEWLINE)
+  f.write(FOURSPACES+"Filter images?:"+filter_images+NEWLINE)
+  f.write(FOURSPACES+"Select the rule criteria:and (extension does isimage) (directory doesnot startwith \".\")")
+  f.write(NEWLINE)
+
+
+def write_metadata():
+  metadata_extraction = params['metadata']['con_metadata_extraction']
+  extract = metadata_extraction['extract']
+
+  if 'extraction_method' in metadata_extraction:
+    method_count = str(len(metadata_extraction['extraction_method']))
+  else:
+    method_count = "1"
+
+  f.write(NEWLINE)
+  f.write("Metadata:[module_num:2|svn_version:\\'Unknown\\'|variable_revision_number:4|show_window:False|notes:\\x5B\\'The Metadata module optionally allows you to extract information describing your images (i.e, metadata) which will be stored along with your measurements. This information can be contained in the file name and/or location, or in an external file.\\'\x5D|batch_state:array(\x5B\x5D, dtype=uint8)|enabled:True|wants_pause:False]"+NEWLINE)
+
+  f.write(FOURSPACES+"Extract metadata?:"+extract+NEWLINE)  
+  
+  if extract=="No":
+    f.write(FOURSPACES+"Metadata data type:Text"+NEWLINE)
+    f.write(FOURSPACES+"Metadata types:{}"+NEWLINE)
+    f.write(FOURSPACES+"Extraction method count:"+method_count+NEWLINE)
+    f.write(FOURSPACES+"Metadata extraction method:Extract from file/folder names"+NEWLINE)
+    f.write(FOURSPACES+"Regular expression to extract from file name:^(?P&lt;Plate>.*)_(?P&lt;Well>\x5BA-P\x5D\x5B0-9\x5D{2})_s(?P&lt;Site>\x5B0-9\x5D)_w(?P&lt;ChannelNumber>\x5B0-9\x5D)"+NEWLINE)
+    f.write(FOURSPACES+"Regular expression to extract from folder name:(?P&lt;Date>\x5B0-9\x5D{4}_\x5B0-9\x5D{2}_\x5B0-9\x5D{2})$"+NEWLINE)
+    f.write(FOURSPACES+"Extract metadata from:All images"+NEWLINE)
+    f.write(FOURSPACES+"Select the filtering criteria:and (file does contain \"\")"+NEWLINE)
+    f.write(FOURSPACES+"Metadata file location:"+NEWLINE)
+    f.write(FOURSPACES+"Match file and image metadata:\x5B\x5D"+NEWLINE)
+    f.write(FOURSPACES+"Use case insensitive matching?:No"+NEWLINE)
+  else:
+    f.write(FOURSPACES+"Metadata data type:"+metadata_extraction['metadata_type']+NEWLINE)
+    f.write(FOURSPACES+"Metadata types:{}"+NEWLINE)
+    f.write(FOURSPACES+"Extraction method count:"+method_count+NEWLINE)    

```

Comment:
If you change this at some point, please also use string replacements like '%s foo %s bar\n' % (one, two)
Then the newline is also not needed anymore ...